src/storage/src/hummock/sstable_store.rs

```diff
@@ -180,17 +207,51 @@ impl SstableStore {
         Ok(())
     }
 
+    pub async fn put_sst_stream(
+        &self,
+        sst_id: HummockSstableId,
+        policy: CachePolicy,
+    ) -> HummockResult<SstableStreamingUploader> {
+        let data_path = self.get_sst_data_path(sst_id);
+        Ok(SstableStreamingUploader {
+            id: sst_id,
+            object_uploader: self.store.streaming_upload(&data_path).await?,
+            blocks: Vec::new(),
+            policy,
+        })
+    }
+
+    /// Finish uploading by providing size footer and metadata.
+    pub async fn finish_put_sst_stream(
+        &self,
+        uploader: SstableStreamingUploader,
+        meta: SstableMeta,
+    ) -> HummockResult<()> {
+        uploader.object_uploader.finish().await?;
+        let sst_id = uploader.id;
+        if let Err(e) = self.put_meta(sst_id, &meta).await {
+            self.delete_sst_data(sst_id).await?;
+            return Err(e);
+        }
+        if let CachePolicy::Fill = uploader.policy {
+            debug_assert!(!uploader.blocks.is_empty());
+            let sst = Sstable::new_with_blocks(sst_id, meta, uploader.blocks).unwrap();
```

Comment:
We will no longer cache data blocks in `Sstable` in meta cache after #4590. I think we can avoid adding `Sstable::new_with_blocks` in this PR. We can just refill meta cache Sstable (without blocks) and refill block cache with data blocks here.